presto-main/src/main/java/com/facebook/presto/sql/relational/optimizer/ExpressionOptimizer.java

```diff
@@ -93,6 +101,30 @@ public RowExpression visitCall(CallExpression call, Void context)
             Signature signature = call.getSignature();
 
             if (signature.getName().equals(CAST)) {
+                if (call.getArguments().get(0) instanceof CallExpression) {
+                    // Optimization for CAST(JSON_PARSE(...) AS ARRAY/MAP)
+                    CallExpression innerCall = (CallExpression) call.getArguments().get(0);
+                    if (innerCall.getSignature().getName().equals("json_parse")) {
```

Comment:
Add a comment to `json_parse` that point to here. If we need to change json_parse in the future, it's going to be very important that people be aware of this.